examples/interfaced_recon/main.cc

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <openvslam/system.h>
+#include <popl.hpp>
+#include <spdlog/spdlog.h>
+#include <yaml-cpp/yaml.h>
+
+#include "cameras/l515.h"
+#include "cameras/zed_native.h"
+#include "modules/slam_module.h"
+#include "modules/tsdf_module.h"
+#include "segmentation/inference.h"
+#include "utils/time.hpp"
+#include "utils/cuda/errors.cuh"
+#include "utils/rotation_math/pose_manager.h"
+#include "utils/config_reader.hpp"
+#include "disinfect_slam/disinfect_slam.h"
+
+void run(const ZEDNative &zed_native, const L515 &l515, std::shared_ptr<DISINFSystem> my_sys) {
+  // initialize TSDF
+
+  std::thread t_slam([&]() {
+    cv::Mat img_left, img_right;
+    while (true) {
+      const int64_t timestamp = zed_native.get_stereo_img(&img_left, &img_right);
+      my_sys->feed_stereo_frame(img_left, img_right, timestamp);
+    }
+  });
+
+  std::thread t_tsdf([&]() {
+    cv::Mat img_rgb, img_depth;
```

Comment:
This should be inside while loop due to the way `tsdf_module.Integrate` was implemented. Otherwise, there will be memory aliasing problem.